### PR TITLE
feat: Remove enrolment statuses and add errocodes by aligning to new structure

### PR DESCRIPTION
**Pull request by alfmagsbidbax**
Create 2 layers of enrollment result reporting.

Enrolmentstatus becomes the main status indicator.
Errocode becomes an in depth additional information field. Giving more exact reason for enrolment failures.

### CODE REVIEW

**Review by olebed — CHANGES_REQUESTED** [review]
The PR name should be a valid commit message.
Please review commit history to figure out what the pattern is.
Pay attention where words starts with a capital character and where not.
Have a look at https://en.wikipedia.org/wiki/Imperative_mood

File: openapi/integrator/token-requestor/partner.yaml [hunk 1]
```diff
@@ -176,16 +176,39 @@ components:
...
         - Accepted
         - Rejected
         - Failed
-        - InvalidFundingPan
-        - CardExpired
-        - CardNotActivated
+
+    ErrorCode:
+      type: string
+      example: Accepted
```
Comment (olebed) [inline openapi/integrator/token-requestor/partner.yaml:186]: There is no error code `Accepted`, right?

File: openapi/integrator/token-requestor/partner.yaml [hunk 2]
```diff
@@ -176,16 +176,39 @@ components:
...
+      description: |
+        Code specifying the result of the operation:
+        * ENROLMENT_OPERATION_FAILED -
```
Comment (olebed) [inline openapi/integrator/token-requestor/partner.yaml:189]: Why duplicating enum values in the description? Enum values itself are self-describing. Do thoses ` -` at the end of some of them have a special meaning?

File: openapi/integrator/token-requestor/partner.yaml [hunk 3]
```diff
@@ -176,16 +176,39 @@ components:
...
+        * UNKNOWN_CARD -
+        * CARD_NOT_ACTIVATED -
+        * CARD_EXPIRED - 
+        * CARD_SUSPENDED - 
+        * CARD_REVOKED -
+        * NO_ELIGIBLE_CARDS_FOUND _
+        * FPAN_PROV_COUNT_EXCEEDED -
+        * INVALID_FUNDING_PAN -
+      enum:
+        - Enrolment_Operation_Failed
```
Comment (olebed) [inline openapi/integrator/token-requestor/partner.yaml:201]: It seems like we use PascalCase for enums, i.e. `Card_Not_Activated` should be `CardNotActivated`?

Comment (alfmagsbidbax) [reply openapi/integrator/token-requestor/partner.yaml:201]: Resolved.

File: openapi/integrator/token-requestor/partner.yaml (hunk 2 above)
Comment (alfmagsbidbax) [reply openapi/integrator/token-requestor/partner.yaml:189]: Resolved.

File: openapi/integrator/token-requestor/partner.yaml (hunk 1 above)
Comment (alfmagsbidbax) [reply openapi/integrator/token-requestor/partner.yaml:186]: Resolved.

File: openapi/integrator/token-requestor/partner.yaml [hunk 4]
```diff
@@ -176,16 +176,39 @@ components:
...
-        * CardExpired - Card has expired.
-        * CardNotActivated - Card is not activated.
       enum:
         - Accepted
         - Rejected
         - Failed
-        - InvalidFundingPan
-        - CardExpired
+
+    ErrorCode:
```
Comment (olebed) [inline openapi/integrator/token-requestor/partner.yaml:184]: New entity `ErrorCode` has been introduced, but not referenced in the doc yet.. Is it by design / what's the plan?
Removal of `InvalidFundingPan` and `CardExpired` from the original enum - is it ok/safe from the backward compatibility point of view during transition period?

Comment (alfmagsbidbax) [reply openapi/integrator/token-requestor/partner.yaml:184]: See internal comment.

**Review by olebed — APPROVED** [review]
LGTM!

**Pull request merged**